src/magneticField/CMZField.cpp

```diff
@@ -0,0 +1,306 @@
+#include "crpropa/magneticField/CMZField.h"
+#include "crpropa/Units.h"
+
+namespace crpropa {
+
+CMZField::CMZField() {
+    useMCField = false;
+    useICField = true;
+    useNTFField = false;
+    useRadioArc = false;
+}
+
+double CMZField::getA(double a1) const {
+    return 4*log(2)/a1/a1;  
+}
+
+double CMZField::getL(double a2) const {
+    return a2/2*log(2);
+}
+
+Vector3d CMZField::BPol(const Vector3d& position,const Vector3d& mid, double B1, double a, double L) const{
+    // cylindircal coordinates
+    Vector3d pos = position - mid;
+    double r = sqrt(pos.x*pos.x + pos.y*pos.y);
+    double phi = std::atan2(pos.y, pos.x);
+
+    double r1 = 1/(1+a*pos.z*pos.z);
+    double Bs = B1*exp(-r1*r/L);
+    double Br = 2*a*pow(r1,3)*r*pos.z*Bs;
```

Comment:
Should be pow_integer<3> or r1*r1*r1 for performance reasons, see: https://github.com/CRPropa/CRPropa3/blob/3ca1ba03729b1e8735ab1dba87e22a57d1c2008e/include/crpropa/Common.h#L55

Reply:
yes I have changed it.

Reply:
The corresponding header file (crpropa/Common.h) was not included, which led to failing builds. Should be fixed now.